pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -98,7 +105,7 @@ def reset(self) -> None:
         # track epoch output
         self._epoch_output = [[] for _ in range(self.batch_loop.num_active_optimizers(self.total_batch_idx))]
 
-        if not self.restarting or self._num_training_batches_reached():
```

Comment:
bad merge? this was recently added by @awaelchli

Reply:
No, it wasn't a bad merge. Without this, restart on validation loop won't work.
But I would have to find a way to make both work together.